From 80ff9cd7db44556eef1e098e183e3f5952b842d0 Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Fri, 16 May 2025 11:28:32 +0300
Subject: [PATCH] [SW-209955] fw-loader: fail if wrong fit provided

At compile time we set HL_DEVICE_TYPE
so we can use it at run time.

Change-Id: Ibce7f993b4065ccb47f7f639ef99245491222819
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 964786d..c5f71c4 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -23,6 +23,7 @@
 CFLAGS += -I $(EMB_COMMON_MOD_DIR)/lib
 CFLAGS += -I $(EMB_COMMON_MOD_DIR)/lib/common
 CFLAGS += -I $(EMB_COMMON_MOD_DIR)/lib/preboot
+CFLAGS += -D HL_DEVICE_TYPE=\"$(BASE_DEV_NAME)\"
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
